pkg/apis/zookeeper/v1beta1/zookeepercluster_types.go

```diff
@@ -374,7 +374,8 @@ type Persistence struct {
 func (p *Persistence) withDefaults() (changed bool) {
 	if !p.VolumeReclaimPolicy.isValid() {
 		changed = true
-		p.VolumeReclaimPolicy = VolumeReclaimPolicyRetain
+		// Default volume reclaim policy should be delete
+		p.VolumeReclaimPolicy = VolumeReclaimPolicyDelete
```

Comment:
Why should the default here be to delete?  This is the opposite behavior of a `StatefulSet`, and is not the expected behavior in ECS Flex (which also uses the ZK operator).

Reply:
@spiegela If the `VolumeReclaimPolicy` is `Retain`, PVCs don't get deleted after the zk cluster is completely deleted (see code in `reconcileFinalizers`), which is not good because if the PVCs keep lying around they cause a problem when cluster is recreated with the same names as these get reattached to the new servers.
Currently to get around this problem, we explicitly set the persistence value in the manifest:
```
apiVersion: "zookeeper.pravega.io/v1beta1"
kind: "ZookeeperCluster"
metadata:
  name: "example"
spec:
  image:
    repository: "pbelgundi/zookeeper"
    tag: "0.2.4"
    pullPolicy: Always
  size: 3
  persistence:
    reclaimPolicy: Delete
```

I don't see why we would want to retain the volumes after deleting the cluster?

Reply:
1. The variable `VolumeReclaimPolicy` does not impact the zookeeper sts in any way. It is simply used to understand the users preference to delete/retain the PVCs after deleting the cluster.
2. k8s does not `auto delete` volumes attached to StatefulSets, when they are deleted because they `may` be needed later. So deleting them is left to user discretion based on the use case. In this case having the PVCs lying around does not serve any purpose and in fact causes issues like the one I mentioned earlier.
3. In case a user wants to resurrect a deleted zookeeper cluster, that is still doable by explicitly setting `persistence: retain` in the manifest. But I don't think most production users would want that.

Reply:
Changed default back to "Retain". With this, zk manifest used by pravega needs to explicitly mention "persistence: Delete" for this fix to take effect.